extension/json/src/functions/table_functions/json_scan.cpp

```diff
@@ -21,6 +23,24 @@ using namespace kuzu::common;
 
 struct JsonScanBindData;
 
+static constexpr idx_t JSON_SCAN_FILE_IDX = 0;
+
+struct JSONWarningSourceData {
+    JSONWarningSourceData() = default;
+    static JSONWarningSourceData constructFrom(const processor::WarningSourceData& warningData) {
```

Comment:
Let's keep the struct body simple.
Can you declare `constructFrom` and implement it after the struct body?

Reply:
updated